command/agent/cache/lease_cache.go

```diff
@@ -0,0 +1,813 @@
+			if err != nil {
+				c.logger.Error("failed to persist index", "error", err)
+				return false, err
+			}
+		}
+
+	case path == vaultPathLeaseRevoke:
+		// TODO: Should lease present in the URL itself be considered here?
+		// Get the lease from the request body
+		jsonBody := map[string]interface{}{}
+		if err := json.Unmarshal(req.RequestBody, &jsonBody); err != nil {
+			return false, err
+		}
+		leaseIDRaw, ok := jsonBody["lease_id"]
+		if !ok {
+			return false, fmt.Errorf("failed to get lease_id from request body")
+		}
+		leaseID, ok := leaseIDRaw.(string)
+		if !ok {
+			return false, fmt.Errorf("expected lease_id the request body to be string")
+		}
+		if err := c.handleCacheClear(ctx, "lease", leaseID); err != nil {
+			return false, err
+		}
+
+	case strings.HasPrefix(path, vaultPathLeaseRevokeForce):
+		// Trim the URL path to get the request path prefix
+		prefix := strings.TrimPrefix(path, vaultPathLeaseRevokeForce)
+		// Get all the cache indexes that use the request path containing the
+		// prefix and cancel the renewer context of each.
+		indexes, err := c.db.GetByPrefix(cachememdb.IndexNameLease, prefix)
+		if err != nil {
+			return false, err
+		}
+
+		_, tokenNSID := namespace.SplitIDFromString(req.Token)
+		for _, index := range indexes {
+			_, leaseNSID := namespace.SplitIDFromString(index.Lease)
+			// Only evict leases that match the token's namespace
+			if tokenNSID == leaseNSID {
+				index.RenewCtxInfo.CancelFunc()
+			}
+		}
+
+	case strings.HasPrefix(path, vaultPathLeaseRevokePrefix):
+		// Trim the URL path to get the request path prefix
+		prefix := strings.TrimPrefix(path, vaultPathLeaseRevokePrefix)
+		// Get all the cache indexes that use the request path containing the
+		// prefix and cancel the renewer context of each.
+		indexes, err := c.db.GetByPrefix(cachememdb.IndexNameLease, prefix)
+		if err != nil {
+			return false, err
+		}
+
+		_, tokenNSID := namespace.SplitIDFromString(req.Token)
+		for _, index := range indexes {
+			_, leaseNSID := namespace.SplitIDFromString(index.Lease)
+			// Only evict leases that match the token's namespace
+			if tokenNSID == leaseNSID {
+				index.RenewCtxInfo.CancelFunc()
+			}
+		}
+
+	default:
+		return false, nil
+	}
+
+	c.logger.Debug("triggered caching eviction from revocation request")
+
+	return true, nil
+}
+
+// deriveNamespaceAndRevocationPath returns the namespace and relative path for
+// revocation paths.
+//
+// If the path contains a namespace, but it's not a revocation path, it will be
+// returned as-is, since there's no way to tell where the namespace ends and
+// where the request path begins purely based off a string.
+//
+// Case 1: /v1/ns1/leases/revoke  -> ns1/, /v1/leases/revoke
+// Case 2: ns1/ /v1/leases/revoke -> ns1/, /v1/leases/revoke
+// Case 3: /v1/ns1/foo/bar  -> root/, /v1/ns1/foo/bar
+// Case 4: ns1/ /v1/foo/bar -> ns1/, /v1/foo/bar
+func deriveNamespaceAndRevocationPath(req *SendRequest) (string, string) {
```

Comment:
I'm guessing this function goes away once https://github.com/hashicorp/vault/pull/6236 gets merged?

Reply:
We will still need this function in order to know what type of revocation request is being proxied.